src/System.Text.Json/src/System/Text/Json/Serialization/PooledBufferWriter.cs

```diff
@@ -101,7 +103,7 @@ public void Dispose()
         private void CheckIfDisposed()
         {
             if (_rentedBuffer == null)
-                ThrowHelper.ThrowObjectDisposedException(nameof(ArrayBufferWriter<T>));
+                ThrowHelper.ThrowObjectDisposedException(nameof(PooledBufferWriter<T>));
```

Comment:
When can this exception actually occur?  And if it can occur and can leak out to callers, since "This is an implementation detail", do we really want the internal type's name in the exception message?

Reply:
This was copy/pasted from the existing API that was going to be public. Given this is totally internal now, we don't need such checks/exceptions (and can be `Debug.Asserts`). We should be using it correctly, internally.